### PR TITLE
ci: add skip directives for long tests

### PR DESCRIPTION
**Pull request by ammario**
This PR introduces many CI optimizations:

1. The `[ci-skip]` PR body directive to skip the Postgres and end to end tests
2. Improved caching that cuts the Go test matrix in half
3. Increasing Go test parallelism for ~20% gains
4. Enable caching in webpack (4x frontend build)

[ci-skip test/go/postgres]

### CODE REVIEW

**Review by kylecarbs — APPROVED** [review]

**Comment by presleyp** [comment]
Arguments for frontend codeowners:

- it makes for an efficient code review cycle in many cases. Someone makes a meaningful change to the frontend, so they want the people who know the codebase best to review it. Those people all get invited to review without anyone having to remember who all is on the team. Those people all get notifications via Slack if they've enabled that. Any one of them can review and fulfill the requirement, so whoever is free first unblocks the PR, rather than the implementer choosing someone who might not be free for a while.
- FE and BE require different areas of expertise, and TS and Go have different coding conventions, so it's natural that code quality will suffer without review from those in the know. The v1 frontend has some misleading and difficult to refactor code and I think this is why.  I don't think FE is special in this way; I would not merge a BE PR without a BE approval. 

Proposal:

1. Make the setting overridable if possible. Grey used to be under the impression that this was already the case. This way, we get the code quality and review efficiency benefits on a "meaty" FE PR, and we can move quickly on a PR that triggers the requirement in a trivial way.
2. Maintain openness to adding to the "FE team." As we do more cross-functional work, more people may find that they have the expertise to function as a FE reviewer, and then we can add them to the list.

**Pull request merged**
